website/client/js/services/userServices.js

```diff
@@ -231,7 +236,49 @@ angular.module('habitrpg')
               var critBonus = crit * 100 - 100;
               Notification.crit(critBonus);
             }
-            if (drop) user._tmp.drop = drop;
+            if (drop) {
+              var text, notes, type;
+              $rootScope.playSound('Item_Drop');
+
+              // Note: For Mystery Item gear, drop.type will be 'head', 'armor', etc
+              // so we use drop.notificationType below.
+
+              if (drop.type !== 'gear' && drop.type !== 'Quest' && drop.notificationType !== 'Mystery') {
+                if (drop.type === 'Food') {
+                  type = 'food';
+                } else if (drop.type === 'HatchingPotion') {
+                  type = 'hatchingPotions';
+                } else {
+                  type = drop.type.toLowerCase() + 's';
+                }
+                if(!user.items[type][drop.key]){
+                  user.items[type][drop.key] = 0;
+                }
+                user.items[type][drop.key]++;
+              }
+
+              if (drop.type === 'HatchingPotion'){
+                text = Content.hatchingPotions[drop.key].text();
+                notes = Content.hatchingPotions[drop.key].notes();
+                Notification.drop(env.t('messageDropPotion', {dropText: text, dropNotes: notes}), drop);
+              } else if (drop.type === 'Egg'){
+                text = Content.eggs[drop.key].text();
+                notes = Content.eggs[drop.key].notes();
+                Notification.drop(env.t('messageDropEgg', {dropText: text, dropNotes: notes}), drop);
+              } else if (drop.type === 'Food'){
+                text = Content.food[drop.key].text();
+                notes = Content.food[drop.key].notes();
+                Notification.drop(env.t('messageDropFood', {dropArticle: drop.article, dropText: text, dropNotes: notes}), drop);
+              } else if (drop.type === 'Quest') {
+                $rootScope.selectedQuest = Content.quests[drop.key];
+                $rootScope.openModal('questDrop', {controller:'PartyCtrl', size:'sm'});
+              } else {
+                // Keep support for another type of drops that might be added
+                Notification.drop(drop.dialog);
+              }
+
+              // Analytics.track({'hitType':'event','eventCategory':'behavior','eventAction':'acquire item','itemName':after.key,'acquireMethod':'Drop'});
```

Comment:
This is the one part I'm not sure how to deal with. If I inject Analytics here, I get a circular dependency: `User -> Analytics -> User`. Can we move this tracking server side?